### PR TITLE
Records NetID of user that accepts/rejects handover

### LINKED ISSUE #24: Harmonize django user model with external auth (shib) and duke ds api
Currently, handovers and drafts can be created in the database with any from_user_id. Practically, it only makes sense to create a handover/draft with the from_user_id set to a known user (for which we have an API key).

With this requirement in place we can prevent clients from creating drafts/handovers if the sender has not registered. Additionally, since the user must authenticate, we don't need/want them to supply their id

### LINKED ISSUE #31: Shibboleth Auth
Right now anyone can accept the handover.
We should add a netid/password prompt(Shibboleth).
Right now we are using nginx.
Based on [these docs](https://wiki.shibboleth.net/confluence/display/SHIB2/Integrating+Nginx+and+a+Shibboleth+SP+with+FastCGI) we would need this module: https://github.com/nginx-shib/nginx-http-shibboleth.

### PR DESCRIPTION
**Pull request by dleehr**
Adds a `performed_by` field to the handover and integrates a shibboleth remote user package. Now, visiting any url under `/ownership` requires login.

The login requirement does not specify the type of account. In the apache config (see `production/handoverservice.conf`), `/ownership` is protected by shibboleth, and this changeset enables a Shibboleth middleware and RemoteUserBackend that create and authenticate Django users based on valid shibboleth logins.

In the development config, `ownership` is not protected by shibboleth, but the views still require login. In this case, django redirects users to login with a local django account.

Since this changeset creates and authenticates django users from any shibboleth login, those users are recorded in the django-simple-history fields. The performed_by field is text though, and won't be cleared if the user record is deleted.

Fixes #31 
Fixes #24

### CODE REVIEW

**Comment by dleehr** [comment]
Some config cleanup reported in #35

**Comment by dleehr** [comment]
d067479 and afd34d4 fix #35

File: handoverservice/urls.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+from django.contrib.auth import views as auth_views
 
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'^ownership/', include('ownership.urls')),
     url(r'^api/v1/', include('handover_api.urls')),
     url(r'^api-auth/', include('rest_framework.urls',
                                namespace='rest_framework')),
+    url(r'^accounts/login/$', auth_views.login, {'template_name': 'ownership/login.html' }),
+    url(r'^accounts/logout/$', auth_views.login, {'template_name': 'ownership/logged_out.html' }),
```
Comment (johnbradley) [inline handoverservice/urls.py]: Shouldn't this be auth_views.logout.

File: ownership/templates/ownership/login.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+<p>Your username and password didn't match. Please try again.</p>
+{% endif %}
+
+{% if next %}
+    {% if user.is_authenticated %}
+    <h4>Your account doesn't have access to this page. To proceed,
+    please login with an account that has access.</h4>
+    {% else %}
+    <h4>Please login to see this page.</h4>
+    {% endif %}
```
Comment (johnbradley) [inline ownership/templates/ownership/login.html]: Having these the same size as the __Login__ header looks a little strange.
Perhaps make them smaller.

Comment (dleehr) [inline ownership/templates/ownership/login.html]: Oh yeah, I did have these as `<p>` tags initially, but then I added the `<h4>` so now everything's huge!

**Comment by johnbradley** [comment]
I was able to run migrations, tests, and the webserver.
I was able to reject and accept a handover and see user logged in. 👍 
I didn't test the shibboleth parts.

Having two methods of authentication is a little confusing, but I understand it is a compromise for increased security. Otherwise LGTM.

**Comment by dleehr** [comment]
> Having two methods of authentication is a little confusing, but I understand it is a compromise for increased security. Otherwise LGTM.

Yeah I thought about that a lot - I followed the lead of the framework. An authenticated user is an authenticated user and we don't care how they authenticated. That makes it easy to test each piece without real shibboleth users (or their passwords). The second method of authentication (shibboleth) really only appears in the docker production config. So if you don't look in the `/production` dir, you'd never see anything about shib.

Do you have any feedback on how it could be less confusing? More documentation or comments? I suppose a README in `/production` would help

File: handoverservice/urls.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
+from django.contrib.auth import views as auth_views
 
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'^ownership/', include('ownership.urls')),
     url(r'^api/v1/', include('handover_api.urls')),
     url(r'^api-auth/', include('rest_framework.urls',
                                namespace='rest_framework')),
+    url(r'^accounts/login/$', auth_views.login, {'template_name': 'ownership/login.html' }),
+    url(r'^accounts/logout/$', auth_views.logout, {'template_name': 'ownership/logged_out.html' }),
```
Comment (johnbradley) [inline handoverservice/urls.py]: A comment here about how these are only used for development would be helpful.

**Comment by johnbradley** [comment]
Outside of a comment about accounts/login and accounts/logout being develop only your comment in
AUTHENTICATION_BACKENDS seems to explain it well enough for me.

**Pull request merged**